pbft/pbft.cpp

```diff
@@ -1596,16 +1684,15 @@ pbft::handle_config_message(const pbft_msg& msg, const std::shared_ptr<pbft_oper
 }
 
 bool
-pbft::move_to_new_configuration(hash_t config_hash)
+pbft::move_to_new_configuration(const hash_t& config_hash)
 {
-    if (this->configurations.is_enabled(config_hash))
-    {
-        this->configurations.set_current(config_hash);
-        this->configurations.remove_prior_to(config_hash);
+    if (this->configurations.current()->get_hash() == config_hash)
         return true;
-    }
 
-    return false;
+    assert(this->configurations.get(config_hash) != nullptr);
+    this->configurations.set_current(config_hash);
+    this->configurations.remove_prior_to(config_hash);
```

Comment:
I think we shouldn't remove old configurations here (only in do_committed, where we are doing so)